apps/desktop-wallet/src/components/DataList.tsx

```diff
@@ -16,28 +16,78 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { Children, isValidElement, ReactNode } from 'react'
 import styled from 'styled-components'
 
-export default styled.div`
-  border: 1px solid ${({ theme }) => theme.border.primary};
-  border-radius: var(--radius-small);
+interface DataListProps {
+  title?: string
+  children: ReactNode
+}
+
+const DataList = ({ children, title }: DataListProps) => {
+  // TODO: If we like this approach, let's create a hook to check children types on runtime?
+  Children.forEach(children, (child) => {
+    if (child && isValidElement(child) && child.type !== DataListRow) {
+      console.error('DataList only accepts DataList.Row as children.')
+    }
+  })
```

Comment:
I don't have the brain capacity to think of this, I need to come back to it.

Reply:
Didn't we say in the past that this was not a good approach or am I remembering wrongly?

Reply:
I don't remember wrongly ;)

https://react.dev/reference/react/Children

<img width="932" alt="image" src="https://github.com/alephium/alephium-frontend/assets/1579899/3d1bbfe3-4161-4e08-b274-59081a4e66f3">

Reply:
I appreciate what you are trying to do here! It's smart! But I am not in favor of using `Children`: https://react.dev/reference/react/Children

It might have unforeseen consequences. For example, in the TransactionDetailsModal, this part of code could generate `false` as a child of DataList, which would trigger the console error (this in fact I am seeing in the console).

<img width="795" alt="image" src="https://github.com/alephium/alephium-frontend/assets/1579899/828d3744-af78-40a9-a5fd-41b68e659a7b">

I would be much more in favor of one of the alternative solutions proposed in the docs, like using an array of objects as a prop: https://github.com/alephium/alephium-frontend/commit/9cb14739d59a9a5b1abc49bcd0b88ff95df3a85b